packages/plugins/eslint-plugin-react-x/src/rules/no-use-context.ts

```diff
@@ -0,0 +1,101 @@
+import { isReactHookCall, isReactHookCallWithNameAlias } from "@eslint-react/core";
+import type { RuleFeature } from "@eslint-react/shared";
+import { getSettingsFromContext } from "@eslint-react/shared";
+import { AST_NODE_TYPES as T } from "@typescript-eslint/types";
+import { compare } from "compare-versions";
+import type { CamelCase } from "string-ts";
+import { isMatching } from "ts-pattern";
+
+import { createRule } from "../utils";
+
+export const RULE_NAME = "no-use-context";
+
+export const RULE_FEATURES = [
+  "CHK",
+  "MOD",
+] as const satisfies RuleFeature[];
+
+export type MessageID = CamelCase<typeof RULE_NAME>;
+
+export default createRule<[], MessageID>({
+  meta: {
+    type: "problem",
+    docs: {
+      description: "disallow the use of 'useContext'",
+      [Symbol.for("rule_features")]: RULE_FEATURES,
+    },
+    fixable: "code",
+    messages: {
+      noUseContext: "In React 19, 'use' is preferred over 'useContext' because it is more flexible.",
+    },
+    schema: [],
+  },
+  name: RULE_NAME,
+  create(context) {
+    const settings = getSettingsFromContext(context);
+    const useContextAlias = new Set<string>();
+
+    if (!context.sourceCode.text.includes("useContext")) {
+      return {};
+    }
+    const { version } = getSettingsFromContext(context);
+    if (compare(version, "19.0.0", "<")) {
+      return {};
+    }
+    return {
+      CallExpression(node) {
+        if (!isReactHookCall(node)) {
+          return;
+        }
+        if (!isReactHookCallWithNameAlias("useContext", context, [...useContextAlias])(node)) {
+          return;
+        }
+        context.report({
+          messageId: "noUseContext",
+          node,
+          fix(fixer) {
+            switch (node.callee.type) {
+              case T.Identifier:
+                return fixer.replaceText(node.callee, "use");
```

Comment:
Will this also replace the import?

```ts
import { foo, bar, useContext } from "react";
↓
import { foo, bar, use } from "react";
```

Reply:
The import will be replaced, but in the following fix: [#r1932852907](https://github.com/Rel1cx/eslint-react/pull/931/files/709b4b7ca790f785674ccae77c159a79e921d34f#r1932852907)